### ISSUE TITLE
In automatic 1.12+ mode, proxy should test/resolve a backend service before adding

### ISSUE THREAD

**Issue by daninthewoods** [vfarcic/docker-flow-proxy#55]
In the example for go-demo, if your service is not attached to the proxy network (ie. you forgot to add --network proxy), then docker-flow will fail with 500 errors and may not recover.  It should test (and output an appropriate message) when a backend service is registered, but is not reachable.

**Comment by vfarcic** [comment]
The proxy does check whether the communication with the service can be established.

Let me explain the current flow of the automation proxy configuration.

1. The swarm-listener detects a new service and send a notification to the proxy
2. The proxy tries to reconfigure itself. It starts by checking whether the service is reachable. If it isn't, it returns status 500 to the swarm-listener.
3. The swarm-listener retries a notification until the response from the proxy is 200 or it reached the maximum number of retries.

In normal cases, the reason proxy cannot establish the communication with the service is because it is still not running. The listener starts sending notifications to the proxy when a service is created, not necessarily when it's running. Swarm needs to pull the image, create a container, and attach it to its networks. Some time passes between the creation of a service and actual instance running. In other words, when everything is set up correctly, it's normal that the first attempt does not work. The second or the third will.

The problem is that the proxy only knows whether the service is running or not. When the service is not running, it does not know whether that's because it will be running soon, or it is not reachable, or ...

Long story short, at the moment, I added a more descriptive message. Not it will say: "Could not reach the service go-demo. Is the service running and connected to the same network as the proxy?" That message is displayed only when the service is not reachable.

I'm not sure what you mean by "may not recover". In my tests under a similar scenario, it simply rejects the request to reconfigure itself when a service is not reachable and returns 500 (now changed with a more descriptive message). The swarm-listener retries the notifications and, if the service continues being unreachable, gives up after a certain number of retries.

Can you let me know what you mean by "may not recover" and share any info you might have that would help me reproduce the case?

**Comment by daninthewoods** [comment]
Appreciate the very long and detailed reply.  I will experiment more with the explanation you gave and report back. Thanks again.
